src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -224,18 +228,24 @@ public void AddTagHelperAttribute(string name, object value)
         }
 
         /// <summary>
-        /// Execute and retrieve the rendered child content asynchronously.
+        /// Execute children asynchronously with the given <paramref name="encoder"/> in scope and return their
+        /// rendered content.
         /// </summary>
+        /// <param name="useCachedResult">If <c>true</c> multiple calls to this method will not cause re-execution
+        /// of child content; cached content will be returned.</param>
+        /// <param name="encoder">
+        /// The <see cref="HtmlEncoder"/> to use. Does not override all HTML encodings which may occur.
```

Comment:
##### Details galore
The following MVC classes store `HtmlEncoder` instances. This PR affects only the first use.

1. `RazorPage` uses an encoder for C# expressions in `string` tag helper attributes and when writing non-`IHtmlContent` C# expressions out.
2. `UrlResolutionTagHelper` uses an encoder to determine the original `string` value of a URL that includes C# expressions. The encoded value flows through.
 1. `LinkTagHelper` and `ScriptTagHelper` use the base `UrlResolutionTagHelper`'s encoder on the calculated URL when `asp-append-version="true"`.
3. `RazorViewEngine` uses an encoder when instantiating `RazorView` instances.
 1. `RazorView` uses that encoder when instantiating `RazorTextWriter` instances and when copying buffered content to an outer `TextWriter`.
 2. `RazorTextWriter` uses that encoder during and sometimes after `Flush[Async]()` operations.
4. `CacheTagHelper` uses an encoder to calculate the cached content. (Can't store the `IHtmlContent` for the body because that may come from a per-request pool.)
5. `DefaultViewComponentHelper` uses an encoder when instantiating `ViewComponentContext` instances.
 1. `ContentViewComponentResult` and `HtmlContentViewComponentResult` use the `ViewComponentContext` encoder when writing out to a non-`HtmlTextWriter` writer. This won't normally occur when the view component is invoked in a `.cshtml` file but can occur after a `Flush[Async]()` or if the view component is invoked directly from an action. (These class are used to handle specific view component return values.)
6. `ViewComponentResult` uses an encoder when writing the content out to the response stream. (This class is used when a view component is invoked directly from an action.)
7. `DefaultHtmlGenerator` uses an encoder in its support of `Html.Encode()`.
8. `HtmlHelper` uses an encoder when generating a `<form>` start tag.
9. `MvcForm` uses an encoder when writing out `FormContext.EndOfFormContent` to a non-`HtmlTextWriter` writer. However the writer should almost-always be a `HtmlTextWriter`.